akka-persistence-typed/src/main/scala/akka/persistence/typed/Retention.scala

```diff
@@ -34,22 +35,26 @@ final case class RetentionCriteria(snapshotEveryNEvents: Long, keepNSnapshots: L
     // Delete old events, retain the latest
     math.max(0, lastSequenceNr - (keepNSnapshots * snapshotEveryNEvents))
   }
+
+  /** Java API. */
+  def withKeepNSnapshots(keepN: Long): RetentionCriteria =
+    copy(keepNSnapshots = keepN)
+
+  /** Java API. */
+  def withDeleteEventsOnSnapshot(): RetentionCriteria =
+    copy(deleteEventsOnSnapshot = true)
 }
```

Comment:
BTW did we decide anything around this being a case class or not? Could make it hard to evolve since `copy` and `unhappy` etc.

Reply:
I turned it into a regular class for the binary compat evolvability reason. Not yet pushed.